api/src/main/java/org/pmiops/workbench/utils/mappers/CommonMappers.java

```diff
@@ -143,9 +143,11 @@ public Long getInitialCreditsExtension(
   }
 
   @Named("getBillingStatus")
-  public BillingStatus getBillingStatus(DbWorkspace dbWorkspace) {
+  public BillingStatus getBillingStatus(
+      DbWorkspace dbWorkspace, @Context InitialCreditsService initialCreditsService) {
     return (isInitialCredits(dbWorkspace.getBillingAccountName(), workbenchConfigProvider.get())
-            && (dbWorkspace.isInitialCreditsExhausted() || dbWorkspace.isInitialCreditsExpired()))
+            && (dbWorkspace.isInitialCreditsExhausted()
+                || initialCreditsService.areUserCreditsExpired(dbWorkspace.getCreator())))
```

Comment:
Use areUserCreditsExpired function instead of workspace field, because bypass logic is in the service.